python/GafferImageTest/ImageAlgoTest.py

```diff
@@ -291,5 +293,50 @@ def testSortChannelNames( self ):
 		# worth documenting )
 		self.assertEqual( GafferImage.ImageAlgo.sortChannelNames( [ "x", "X", "x.x", "X.X", "X.x", "x.X" ] ), ["X", "x", "X.X", "X.x", "x.X", "x.x"] )
 
+		# Test that our natural sort handles numbers
+		self.assertEqual(
+			GafferImage.ImageAlgo.sortChannelNames( [ "12", "4", "1", "14", "2", "3", "13", "99" ] ),
+			[ "1", "2", "3", "4", "12", "13", "14", "99" ]
+		)
+
+		self.assertEqual(
+			GafferImage.ImageAlgo.sortChannelNames( [ str( 999 - i ) for i in range( 1000 ) ] ),
+			[ str( i ) for i in range( 1000 ) ]
+		)
+
+		# And numbers with varying padding
+		self.assertEqual(
+			GafferImage.ImageAlgo.sortChannelNames( [ "03", "00012", "4", "01", "0000014", "2", "00003", "13", "3", "99" ] ),
+			[ "01", "2", "3", "03", "00003", "4", "00012", "13", "0000014", "99" ]
+		)
+
+		# Test large numbers ( larger than the size of an integer )
+		largeNums = [ str( ( i % 9 ) + 1 ) + "0" * i for i in range( 30 ) ]
+		self.assertEqual( GafferImage.ImageAlgo.sortChannelNames( largeNums ), largeNums )
```

Comment:
Sorry, this is a comment for something I should have noticed in a previous review : the binding for `sortChannelNames()` doesn't match the C++ semantics. In C++, we're operating in-place and returning nothing, and in Python we're operating on a copy and returning it. The latter seems better matched to a function called `sortedChannelNames().